### ISSUE TITLE
Unable to launch Microsoft Edge debugger

### ISSUE THREAD

**Issue by mkhoatd** [expo/expo#21910]
### Summary

When I press j to open debugger, it show error
```
Error: [LaunchBrowser] Unable to find a browser on the host to open the inspector. Supported browsers: Google Chrome, Microsoft Edge
```
The error happen in both Powershell 7 and Windows Powershell

When I go to `http://localhost:19000/debugger-ui/` there is nothing in the console

![image](https://user-images.githubusercontent.com/75836068/228893793-bcbcd0b5-dc8c-4f73-bbae-aaff5909fc40.png)

Here is my execution policy:

- Powershell 7
```
        Scope ExecutionPolicy
        ----- ---------------
MachinePolicy       Undefined
   UserPolicy       Undefined
      Process       Undefined
  CurrentUser          Bypass
 LocalMachine    RemoteSigned
```
- Windows Powershell
```
        Scope ExecutionPolicy
        ----- ---------------
MachinePolicy       Undefined
   UserPolicy       Undefined
      Process       Undefined
  CurrentUser          Bypass
 LocalMachine    Unrestricted
```

### What platform(s) does this occur on?

Android

### SDK Version

48.0.5

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.22621
    Binaries:
      Node: 18.15.0 - C:\Program Files\nodejs\node.EXE
      npm: 9.5.0 - C:\Program Files\nodejs\npm.CMD
    SDKs:
      Android SDK:
        API Levels: 31, 32, 33, 33
        Build Tools: 30.0.3, 33.0.0, 33.0.1, 33.0.2
        System Images: android-30 | Google APIs Intel x86 Atom, android-33 | Google APIs Intel x86 Atom_64
    npmPackages:
      expo: ~48.0.5 => 48.0.5 
      react: 18.2.0 => 18.2.0 
      react-native: 0.71.3 => 0.71.3 
      react-navigation: ^4.4.4 => 4.4.4 
    Expo Workflow: managed
```

### Minimal reproducible example

-  Run `npx expo start ` in Windows Powershell or Powershell 7
-  Press j

**Comment by Kudo** [comment]
hi there! this is duplicated of #20255. i have a proposing fix in #20342. could you help to apply the following patch in `node_modules/@expo/dev-server/build/LaunchBrowserImplWindows.js` and see whether it works for you?

```diff
@@ -46,6 +46,7 @@ const IS_WSL = require('is-wsl') && !require('is-docker')();
 class LaunchBrowserImplWindows {
   constructor() {
     _defineProperty(this, "_appId", void 0);
+    _defineProperty(this, "_powershellEnv", void 0);
     _defineProperty(this, "MAP", {
       [_LaunchBrowser().LaunchBrowserTypes.CHROME]: {
         appId: 'chrome',
@@ -60,9 +61,11 @@ class LaunchBrowserImplWindows {
   async isSupportedBrowser(browserType) {
     let result = false;
     try {
+      const env = await this.getPowershellEnv();
       const {
         status
       } = await (0, _spawnAsync().default)('powershell.exe', ['-c', `Get-Package -Name '${this.MAP[browserType].fullName}'`], {
+        env,
         stdio: 'ignore'
       });
       result = status === 0;
@@ -98,13 +101,31 @@ class LaunchBrowserImplWindows {
         // And we cannot just call `process.kill()` kill it.
         // The implementation tries to find the pid of target chromium browser process (with --app=https://chrome-devtools-frontend.appspot.com in command arguments),
         // and uses taskkill to terminate the process.
+        const env = await this.getPowershellEnv();
         await (0, _spawnAsync().default)('powershell.exe', ['-c', `taskkill.exe /pid @(Get-WmiObject Win32_Process -Filter "name = '${this._appId}.exe' AND CommandLine LIKE '%chrome-devtools-frontend.appspot.com%'" | Select-Object -ExpandProperty ProcessId)`], {
+          env,
           stdio: 'ignore'
         });
       } catch {}
       this._appId = undefined;
     }
   }
+
+  /**
+   * This method is used to get the powershell environment variables for `Get-Package` command.
+   * Especially for powershell 7, its default `PSModulePath` is different from powershell 5 and `Get-Package` command is not available.
+   * We need to set the PSModulePath to include the default value of powershell 5.
+   */
+  async getPowershellEnv() {
+    if (this._powershellEnv) {
+      return this._powershellEnv;
+    }
+    const PSModulePath = (await (0, _spawnAsync().default)('powershell.exe', ['-c', 'echo "$PSHOME\\Modules"'])).stdout.trim();
+    this._powershellEnv = {
+      PSModulePath
+    };
+    return this._powershellEnv;
+  }
 }
 exports.default = LaunchBrowserImplWindows;
 //# sourceMappingURL=LaunchBrowserImplWindows.js.map
\ No newline at end of file
```

**Comment by mkhoatd** [comment]
I got this error after that
`TypeError: _open(...).default.openApp is not a function`

**Comment by Kudo** [comment]
that's strange! could you share the entire content of the `LaunchBrowserImplWindows.js`?

**Comment by Kudo** [comment]
thanks @mkhoatd! that's strange to me tbh, since we don't touch the code near `_open(...).default.openApp`. if unappling the patch, does it work for you?

**Comment by matart15** [comment]
I use WSL2 + ios device for development. I can't launch browser from inside WSL. 

Currently i use [react-native-debugger](https://github.com/jhen0409/react-native-debugger) for network ispection.

but Expo 49 beta [changed log](https://blog.expo.dev/expo-sdk-49-beta-is-now-available-6373e78f8624)  stated `Remote debugging for JSC is disabled in Expo Go and expo-dev-client`,  which means we won't able to use the [react-native-debugger](https://github.com/jhen0409/react-native-debugger).

**Comment by welshy557** [comment]
@Kudo I see this change was implemented and I see it in my node_modules. However, I'm currently unable to open the debugger. I'm also unable to go to http://localhost:19000/debugger-ui. It just says refused to connect. Whenever I start expo with --tunnel, I get the same error when I press J. However, I can go to the URL provided with /debugger-ui, and it does open however, it just says "Status: Waiting, press Ctrl R in the simulator to reload and connect.". Reloading does not resolve the issue. Any suggestions?

**Comment by pgk1216** [comment]
@welshy557 Did you end up figuring it out? I'm having the same issue

**Comment by ShaneKeney** [comment]
> I use WSL2 + ios device for development. I can't launch browser from inside WSL. It gives same error message.
> 
> Currently on expo 48, i use [react-native-debugger](https://github.com/jhen0409/react-native-debugger) for network ispection.
> 
> but Expo 49 beta [changed log](https://blog.expo.dev/expo-sdk-49-beta-is-now-available-6373e78f8624) stated `Remote debugging for JSC is disabled in Expo Go and expo-dev-client`, which means we won't able to use the [react-native-debugger](https://github.com/jhen0409/react-native-debugger).

@matart15 Have you figured out how to connect the debugger to the Expo app instance on WSL2?  I use the same development setup and am looking into how to solve this currently so I have have access to Chrome Devtools debugging tools.

**Comment by matart15** [comment]
> Have you figured out how to connect the debugger to the Expo app instance on WSL2? 
Sorry I could not solve that problem. on wsl development I was mostly looking at terminal logs. Which is terrible.

Now I moved to my old mac. My old mac specs are bad but DX is better.
